### PR TITLE
APEXMALHAR-2153 Adding user docs for POJOEnricher.

### PR DESCRIPTION
**Pull request by chinmaykolhatkar**
@bhupeshchawda Please review and merge.

### CODE REVIEW

File: docs/operators/enricher.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
+POJO Enricher
+=============
+
+## Operator Objective
+This operator receives an POJO ([Plain Old Java Object](https://en.wikipedia.org/wiki/Plain_Old_Java_Object)) as an incoming tuple and uses an external source to enrich the data in 
+the incoming tuple and finally emits the enriched data as a new enriched POJO.
+
+POJOEnricher supports enrichment from following external sources:
+
+1. **JSON File Based** - Reads the file in memory having content stored in JSON format and use that to enrich the data. This can be done using FSLoader implementation.
```
Comment (bhupeshchawda) [inline docs/operators/enricher.md]: Should this be small enough to fit in memory. If so, please mention.

File: docs/operators/enricher.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+## Operator Objective
+This operator receives an POJO ([Plain Old Java Object](https://en.wikipedia.org/wiki/Plain_Old_Java_Object)) as an incoming tuple and uses an external source to enrich the data in 
+the incoming tuple and finally emits the enriched data as a new enriched POJO.
+
+POJOEnricher supports enrichment from following external sources:
+
+1. **JSON File Based** - Reads the file in memory having content stored in JSON format and use that to enrich the data. This can be done using FSLoader implementation.
+2. **JDBC Based** - Any JDBC store can act as an external entity to which enricher can request data for enriching incoming tuples. This can be done using JDBCLoader implementation.
+
+POJO Enricher does not hold any state and is **idempotent**, **fault-tolerance** and **statically/dynamically partitionable**.
```
Comment (bhupeshchawda) [inline docs/operators/enricher.md]: nit: tolerant*

File: docs/operators/enricher.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+## Operator Usecase
+1. Bank ***transaction records*** usually contains customerId. For further analysis of transaction one wants the customer name and other customer related information. 
+Such information is present in another database. One could enrich the transaction's record with customer information using POJOEnricher.
+2. ***Call Data Record (CDR)*** contains only mobile/telephone numbers of the customer. Customer information is missing in CDR. POJO Enricher can be used to enricher
```
Comment (bhupeshchawda) [inline docs/operators/enricher.md]: nit: can be used to enrich*

File: docs/operators/enricher.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    * Operator: ***[com.datatorrent.contrib.enrich.POJOEnricher](https://www.datatorrent.com/docs/apidocs/com/datatorrent/contrib/enrich/POJOEnricher.html)***
+    * FSLoader: ***[com.datatorrent.contrib.enrich.FSLoader](https://www.datatorrent.com/docs/apidocs/com/datatorrent/contrib/enrich/FSLoader.html)***
+    * JDBCLoader: ***[com.datatorrent.contrib.enrich.JDBCLoader](https://www.datatorrent.com/docs/apidocs/com/datatorrent/contrib/enrich/JDBCLoader.html)***
+
+## Properties, Attributes and Ports
+### <a name="props"></a>Properties of POJOEnricher
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *includeFields* | List of fields from database that needs to be added to output POJO. | List<String\> | Yes | N/A |
+| *lookupFields* | List of fields from input POJO which will form a *unique composite* key for querying to database | List<String\> | Yes | N/A |
```
Comment (bhupeshchawda) [inline docs/operators/enricher.md]: querying database `or the JSON file?`

Comment (bhupeshchawda) [inline docs/operators/enricher.md]: Will composite key work for JSON?

File: docs/operators/enricher.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+| **Port** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *input* | Tuple which needs to be enriched are received on this port | Object (POJO) | Yes |
+| *output* | Tuples that are enriched from external source are emitted from on this port | Object (POJO) | No |
+
+## Limitations
+Current POJOEnricher contains following limitation:
+
+1. FSLoader loads the file content in memory. Though it loads only the composite key and composite value in memory, a very large amount of data would bloat the memory and make the operator go OOM. In case the filesize is large, allocate sufficient memory to the POJOEnricher.
+2. Incoming POJO should be a subset of outgoing POJO.
```
Comment (bhupeshchawda) [inline docs/operators/enricher.md]: Is this necessary?

File: docs/operators/enricher.md (hunk 1 above)
Comment (chinmaykolhatkar) [inline docs/operators/enricher.md]: I've given traits on being in memory later. I think this is just a starting text. And should be enough for this.

File: docs/operators/enricher.md (hunk 4 above)
Comment (chinmaykolhatkar) [inline docs/operators/enricher.md]: Made querying to store.

Comment (chinmaykolhatkar) [inline docs/operators/enricher.md]: yes composite keys work with JSON.

File: docs/operators/enricher.md (hunk 5 above)
Comment (chinmaykolhatkar) [inline docs/operators/enricher.md]: Yes.. That's necessary.

**Pull request merged**